engine/engine.go

```diff
@@ -147,16 +147,16 @@ func (e *Engine) RegisterCallback(cb provider.Callback) {
 	e.cb = cb
 }
 
-func (e *Engine) NotifyPut(ctx context.Context, key provider.LookupKey, metadata stiapi.Metadata) (cid.Cid, error) {
-	log.Debugf("NotifyPut for lookup key")
+func (e *Engine) NotifyPut(ctx context.Context, contextId []byte, metadata stiapi.Metadata) (cid.Cid, error) {
```

Comment:
nit: I think most people will expect the spelling to be contextID, just like we type DoHTTP instead of DoHttp.